### ISSUE TITLE
Give AutomaticallyCrossCertifiedUsers More Rights on names.nsf

### ISSUE THREAD

**Issue by JoelProminic** [DominoVagrant/demo#13]
I have seen some use cases recently where I need to make more modifications to names.nsf:
- Configure Directory Assistance
- Setup a Internet Sites document for setting CORS headers

However, I couldn't make these changes because my cross-certified ID didn't have sufficient access to names.nsf.   I think this is mainly missing the roles.     I'd like to get this updated so that this server can be used as a sandbox.

**Comment by JoelProminic** [comment]
I tried this: 

```
	"appConfiguration": {
		"databases": [
		  {
			"filePath": "names.nsf",
			"action": "update",
			"ACL": {
			  "ACLEntries": [
			    {
			      "name": "AutomaticallyCrossCertifiedUsers",
			      "level": "manager",
			      "type": "personGroup",
			      "isPublicReader": true,
			      "isPublicWriter": true,
			      "canDeleteDocuments": true,
			      "canReplicateOrCopyDocuments": true,
			      "roles": [
			      	"GroupCreator",
			      	"GroupModifier",
			      	"NetCreator",
			      	"PolicyCreator",
			      	"PolicyModifier",
			      	"PolicyReader",
			      	"NetModifier ",
			      	"ServerCreator",
			      	"ServerModifier",
			      	"UserCreator",
			      	"UserModifier"
			      ]
			    }
		  	  ]
			},

```

However, the ACL Entry for AutomaticallyCrossCertifiedUsers was not created in this case.  I confirmed it was being created in previous executions.  I'm doing more tests to debug my configuration.

**Comment by JoelProminic** [comment]
We are still having problems with getting the AutomaticallyCrossCertifiedUsers group to work properly (my user is not detected as a member of the group), so that the user can administrate names.nsf without using the admin ID.

I found that the problem can be worked around by adding an explicit entry for the cross-certified user's name.  We could like to update CrossCertifyNotesID.java to update the names.nsf ACL.   Here are some [examples](https://github.com/DominoGenesis/Genesis/blob/bde62b70bcd0fef35c41117a87daedba4b80a6f6/src/main/java/net/prominic/genesis/JSONRules.java#L534).

We are planning to submit a ticket to HCL for the AutomaticallyCrossCertifiedUsers issue, so we may disable this change if we can get the group working.

**Comment by JoelProminic** [comment]
I printed the available names.nsf roles from the Java API:

    Valid ACL Roles:  
      [UserCreator]
      [UserModifier]
      [GroupCreator]
      [GroupModifier]
      [NetCreator]
      [NetModifier]
      [ServerCreator]
      [ServerModifier]
      [PolicyCreator]
      [PolicyModifier]
      [PolicyReader]
      [DenyAccessRead]

And I confirmed this JSON configuration worked with the Genesis snippet above.  This is working well in the CrossCertifyNotesID application now.

```
{
  "level": "manager",
  "type": "person",
  "canDeleteDocuments": true,
  "canReplicateOrCopyDocuments": true,
  "roles": [ 
    "[GroupCreator]",
    "[GroupModifier]",
    "[NetCreator]",
    "[PolicyCreator]",
    "[PolicyModifier]",
    "[PolicyReader]",
    "[NetModifier]",
    "[ServerCreator]",
    "[ServerModifier]",
    "[UserCreator]",
    "[UserModifier]"
  ]
}
```

However, if I use the same format with the One-Touch Setup (setup.json), I get an error like this:

    domino-vm: Error: ACL entry specified role [GroupCreator] that does not exist in the ACL

**Comment by JoelProminic** [comment]
I got the setup.json configuration to work by removing the square brackets.

My user is still not recognized as a member of "AutomaticallyCrossCertifiedUsers", but we are doing more investigation for this.   I noticed that it did start working after we cross-certified "/PNI" in addition to the individual ID.   Doug doesn't think this explains the original problem, but I was able to reproduce this result on a fresh instance.

I'm closing this issue, and I'll talk to @MarkProminic about adding it to `demo-tasks`
